### PR TITLE
feat(server): decrement placed quantities on order canceled

### PR DESCRIPTION
**Pull request by BOCOVO**
Decrement placed quantities on availability product when an order is canceled.

[screen-capture (43).webm](https://github.com/user-attachments/assets/8833b9b9-631e-4aaa-8101-f1c08f96a353)

### CODE REVIEW

File: src/services/availability.ts [hunk 1]
```diff
@@ -207,6 +213,43 @@ class AvailabilityService extends TransactionBaseService {
       success: !!updateResult.affected,
     };
   }
+
+  async handleRollbackPlacedQuantities(em: EntityManager, orderID: string) {
```
Comment (Valentino-Houessou) [inline src/services/availability.ts:217]:
```suggestion
  async rollbackPlacedQuantities(em: EntityManager, orderID: string) {
```

**Review by Valentino-Houessou — CHANGES_REQUESTED** [review]
typo suggestions

File: src/subscribers/decrement-placed-qty.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+type OrderCancelEvent = {
+  id: string;
+};
+
+export default async function orderCanceledHandler({
+  data,
+  container,
+}: SubscriberArgs<OrderCancelEvent>) {
+  const orderService: AvailabilityService = container.resolve(
```
[the remark this answers is not in the document]
Comment (BOCOVO) [reply src/subscribers/decrement-placed-qty.ts:16]: bien vu @Valentino-Houessou

**Comment by BOCOVO** [comment]
@Valentino-Houessou 
tu peux repasser ici stp ?

**Review by Valentino-Houessou — APPROVED** [review]

**Pull request merged**